app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -524,6 +522,33 @@ def traces_helpers(search:)
         ::Katello::HostTracer.helpers_for(traces)
       end
 
+      def probably_rhel?
+        # Get the os name from sub-man facts rather than operatingsystem. This is
+        # less likely to have been changed by the user.
+        os_name = facts['distribution::name']
+        # if this fact isn't there, we can ignore it because the host is not "managed"
+        os_name.present? && os_name.start_with?('Red Hat Enterprise Linux')
+      end
+
+      def rhel_eos_schedule_index
+        return nil unless probably_rhel?
+        major = operatingsystem&.major
```

Comment:
What if `major` is `nil`? You'll end up with just `RHEL` as the name. If `operatingsystem` can't be `nil` then just `operatingsystem.major` should suffice, right?